src/ripple/app/main/LoadManager.cpp

```diff
@@ -32,38 +33,50 @@
 
 namespace ripple {
 
-class LoadManagerImp
-    : public LoadManager
-    , public beast::Thread
+class LoadManagerImp : public LoadManager
 {
 public:
     //--------------------------------------------------------------------------
 
-    beast::Journal m_journal;
-    using LockType = std::mutex;
-    using ScopedLockType = std::lock_guard <LockType>;
-    LockType mLock;
+    beast::Journal mJournal;
+    std::thread mThread;
 
-    bool mArmed;
+    std::mutex mMutex;          // Guards mDeadLock and mArmed
 
     int mDeadLock;              // Detect server deadlocks
+    bool mArmed;
+
     //--------------------------------------------------------------------------
 
     LoadManagerImp (Stoppable& parent, beast::Journal journal)
         : LoadManager (parent)
-        , Thread ("loadmgr")
-        , m_journal (journal)
-        , mArmed (false)
+        , mJournal (journal)
+        , mThread ()
+        , mMutex ()
         , mDeadLock (0)
+        , mArmed (false)
     {
         UptimeTimer::getInstance ().beginManualUpdates ();
     }
 
-    ~LoadManagerImp ()
+    LoadManagerImp () = delete;
+    LoadManagerImp (LoadManagerImp const&) = delete;
+    LoadManagerImp& operator=(LoadManager const&) = delete;
+
+    ~LoadManagerImp () override
     {
-        UptimeTimer::getInstance ().endManualUpdates ();
+        try
+        {
+            UptimeTimer::getInstance ().endManualUpdates ();
 
-        stopThread ();
+            if (mThread.joinable())
+                onStop ();
```

Comment:
Excellent point.  Thanks for thinking it through better than I did.

I don't believe this destructor should call `stopAsync()`.  This is too low a level to be making that decision which would affect all `Stoppable`s.  I think what I really need to do is add a member `stopping` flag (which is effectively what `beast::Thread` contained).  Then that flag can be used by both the destructor and `onStop()`.

FWIW, I believe having the destructor run before `onStop()` is called is an extraordinary circumstance.  That's why my manual testing didn't find the problem you spotted.